src/io/syncfile.c

```diff
@@ -451,12 +452,9 @@ MVMObject * MVM_file_open_fh(MVMThreadContext *tc, MVMString *filename, MVMStrin
     MVM_free(fmode);
 
     /* Try to open the file. */
-#ifdef _WIN32
-    flag |= _O_BINARY;
-#endif
-    if ((fd = open((const char *)fname, flag, DEFAULT_MODE)) == -1) {
+    if ((fd = uv_fs_open(NULL, &req, (const char *)fname, flag, DEFAULT_MODE, NULL)) < 0) {
         char *waste[] = { fname, NULL };
-        const char *err = strerror(errno);
+        const char *err = uv_strerror(req.result);
```

Comment:
Should this be included in `waste`?

Reply:
Auditing the use of `MVM_exception_throw_adhoc_free` I did not find that to be the case. If it is then there are many exceptions that can be fixed.